pandas/tests/tseries/offsets/test_custom_business_day.py

```diff
@@ -78,7 +80,7 @@ def test_weekmask_and_holidays(self):
 
     @pytest.mark.filterwarnings("ignore:Non:pandas.errors.PerformanceWarning")
     def test_calendar(self):
-        calendar = USFederalHolidayCalendar()
```

Comment:
So was this just getting ignored before? If not and the behavior was working as expected, we should continue supporting these arguments

Reply:
Right, I agree in principle, but with the modified docs and tests in this PR we have some testing with `AbstractHolidayCalendar` and it's subclasses, so this PR would introduce breaking behavior.

Therefore, I would suggest first investigating if those docs/tests are producing correct behavior. If so, I would suggest calling the original issue a bug that needs fixing, but it would still be good to add validation to allow `calendar: None | np.businesscalendar | AbstractHolidayCalendar`